Create postgresql-high-number-of-indexes-check.md (#81)
percona-platform/portal-doc: docs/advisors/checks/postgresql-high-number-of-indexes-check.md

# Check for relations with high number of indexes

## Description

This check will scan all databases in the cluster and list relations with more than 10 indexes.

## Resolution

> Connect to the database and run the query:

Reviewer: ```suggestion
Connect to the database and run the following query:
```

> Connecting to a database and running the query:

Reviewer 2: ```suggestion
Connect to the database and run the query:
```

```
SELECT current_database() AS datname, relname, count(*) AS idxcount FROM pg_stat_user_indexes GROUP BY 1,2 ORDER BY 3 DESC
```

> Relations with a high number of indexes may or not be needed. Therefore, it is advisable to check if the indexes are in use. Having unnecessary indexes can result in unwanted performance issues and overhead due to the index maintenance.  

Reviewer: ```suggestion
Relations with a high number of indexes may or not be needed. Therefore, we recommend checking if the indexes are in use. Unnecessary indexes can cause performance issues and overhead due to index maintenance.  
```

> Relations with a high number of indexes may or not be needed. Therefore, it is advisable to check if the indexes are being used or not. Having unnecessary indexes can result in unwanted performance issues and overhead due to the maintenance of the index.  

Reviewer 2: ```suggestion
Relations with a high number of indexes may or not be needed. Therefore, it is advisable to check if the indexes are in use. Having unnecessary indexes can result in unwanted performance issues and overhead due to the index maintenance.  
```
